src/main/java/org/codehaus/plexus/archiver/jar/JarToolModularJarArchiver.java

```diff
@@ -147,23 +155,42 @@ protected void postCreateArchive() throws ArchiverException {
     private void fixLastModifiedTimeZipEntries() throws IOException {
         long timeMillis = getLastModifiedTime().toMillis();
         Path destFile = getDestFile().toPath();
-        Path tmpZip = Files.createTempFile(destFile.getParent(), null, null);
-        try (ZipFile zipFile = new ZipFile(getDestFile());
-                ZipOutputStream out = new ZipOutputStream(Files.newOutputStream(tmpZip))) {
-            Enumeration<? extends ZipEntry> entries = zipFile.entries();
-            while (entries.hasMoreElements()) {
-                ZipEntry entry = entries.nextElement();
-                // Not using setLastModifiedTime(FileTime) as it sets the extended timestamp
-                // which is not compatible with the jar tool output.
-                entry.setTime(timeMillis);
-                out.putNextEntry(entry);
-                if (!entry.isDirectory()) {
-                    IOUtil.copy(zipFile.getInputStream(entry), out);
+        FileAttribute<?>[] attributes;
+        if (IS_POSIX) {
```

Comment:
If `PosixFileAttributeView` is not available for some reason, `getFileAttributeView` would return `null`q so we can use that to check instead of the default file system.

Reply:
done

Reply:
I think this was taken care by the `IS_POSIX` check because `destFile` was a `Path` instance constructed out of a `File` instance using [File#toPath()](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#toPath--) and this would have been using the default filesystem.
But I followed your suggestion and replaced the `IS_POSIX` check with a `null` check.